text/0670-aws-applicationsignals-enablement-l2.md

```diff
@@ -0,0 +1,439 @@
+# AWS CloudWatch Application Signals L2 Construct for Simplifying Enablement
+
+* **Original Author(s):**: @bjrara
+* **Tracking Issue**: [#670](https://github.com/aws/aws-cdk-rfcs/issues/670)
+* **API Bar Raiser**: @moelasmar
+
+The application-signals-integration module is a collection of L2 constructs which leverages native L1 CFN resources, simplifying the enablement steps
+and the creation of Application Signals resources. The new L2 construct for enablement addresses key challenges in the current ECS CDK enablement
+process, which requires complex manual configurations, including new container setup, volume management, and extensive environment variable configuration.
+
+## Working Backwards
+
+### CHANGELOG
+
+`feat(cloudwatch): introduce Application Signals L2 construct for simplifying the enablement steps.`
+
+### README
+
+#### ApplicationSignalsTaskDefinition
+
+`ApplicationSignalsTaskDefinition` is a construct to update an existing TaskDefinition with Application Signals auto-instrumentation configurations.
+
+|Name |Type |Description |
+|--- |--- |--- |
+|taskDefinition |[ecs.TaskDefinition](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ecs.TaskDefinition.html) |The task definition to render. |
+|instrumentation |[InstrumentationProps](https://quip-amazon.com/OdXQA04nJ3Fv#temp:C:TaA589601f465f847c4aadd74516) |The auto-instrumentation configuration. |
+|serviceName? |string |The serivce identifier in Application Signals.<br>Default: the input task definition name. |
+|overrideEnvironments? |[EnvironmentExtension](https://quip-amazon.com/OdXQA04nJ3Fv#temp:C:TaA4e452437b4f9453183557666b)[] |The Application Signals reserved envrionment variables to overwrite on the main container. |
+|cloudWatchAgent? |[CloudWatchAgentProps](https://quip-amazon.com/OdXQA04nJ3Fv#temp:C:TaA7a894d0dca304d6f818a5d62c) |The CloudWatch agent sidecar configuration.<br>Default: enables a basic agent sidecar container with latest public image. |
```

Comment:
Nit: the 3 links here and some other links in the doc point to internal docs. Should be changed to point to sections in the RFC

Reply:
Thanks! Updated the link!